certora/applyMetaMorphoMunging.patch

```diff
@@ -0,0 +1,143 @@
+ import {MorphoBalancesLib} from "../lib/morpho-blue/src/libraries/periphery/MorphoBalancesLib.sol";
+@@ -45,9 +44,7 @@
+     using UtilsLib for uint256;
+     using SafeCast for uint256;
+     using SafeERC20 for IERC20;
+-    using MorphoLib for IMorpho;
+     using SharesMathLib for uint256;
+-    using MorphoBalancesLib for IMorpho;
+     using MarketParamsLib for MarketParams;
+     using PendingLib for MarketConfig;
+     using PendingLib for PendingUint192;
+@@ -56,7 +53,7 @@
+     /* IMMUTABLES */
+ 
+     /// @inheritdoc IMetaMorphoBase
+-    IMorpho public immutable MORPHO;
++    IMorphoHarness public immutable MORPHO;
+ 
+     /// @inheritdoc IMetaMorphoBase
+     uint8 public immutable DECIMALS_OFFSET;
+@@ -105,6 +102,19 @@
+     /// @inheritdoc IMetaMorphoBase
+     uint256 public lastTotalAssets;
+ 
++    // HARNESS
++    // The index of the identifier of the last market withdrawn.
++    uint256 public lastIndexWithdraw;
++
++    // HARNESS
++    // The rank of a market identifier in the withdraw queue.
++    // Returns 0 if the corresponding market is not in the withdraw queue.
++    mapping(Id => uint256) public withdrawRank;
++
++    // HARNESS
++    // The last index at which a market identifier has been removed from the withdraw queue.
++    mapping(Id => uint256) public deletedAt;
++
+     /* CONSTRUCTOR */
+ 
+     /// @dev Initializes the contract.
+@@ -124,7 +134,7 @@
+     ) ERC4626(IERC20(_asset)) ERC20Permit(_name) ERC20(_name, _symbol) Ownable(owner) {
+         if (morpho == address(0)) revert ErrorsLib.ZeroAddress();
+ 
+-        MORPHO = IMorpho(morpho);
++        MORPHO = IMorphoHarness(morpho);
+         DECIMALS_OFFSET = uint8(uint256(18).zeroFloorSub(IERC20Metadata(_asset).decimals()));
+ 
+         _checkTimelockBounds(initialTimelock);
+@@ -336,6 +346,9 @@
+             seen[prevIndex] = true;
+ 
+             newWithdrawQueue[i] = id;
++
++            // HARNESS
++            withdrawRank[id] = i + 1;
+         }
+ 
+         for (uint256 i; i < currLength; ++i) {
+@@ -353,6 +366,10 @@
+                     }
+                 }
+ 
++                // HARNESS
++                deletedAt[id] = i;
++                delete withdrawRank[id];
++
+                 delete config[id];
+             }
+         }
+@@ -585,7 +602,11 @@
+     /// @inheritdoc IERC4626
+     function totalAssets() public view override returns (uint256 assets) {
+         for (uint256 i; i < withdrawQueue.length; ++i) {
+-            assets += MORPHO.expectedSupplyAssets(_marketParams(withdrawQueue[i]), address(this));
++            uint256 supplyShares =
++                IMorphoHarness(MORPHO).position(_marketParams(withdrawQueue[i]).id(), address(this)).supplyShares;
++            (uint256 totalSupplyAssets, uint256 totalSupplyShares,,) =
++                MorphoBalancesLib.expectedMarketBalances(MORPHO, _marketParams(withdrawQueue[i]));
++            assets += supplyShares.toAssetsDown(totalSupplyAssets, totalSupplyShares);
```

Comment:
This essentially removes the reason why metamorpho was munged in the first place, why not just using Metamorpho directly (without munging it) ?